rotkehlchen/chain/evm/tokens.py

```diff
@@ -220,13 +249,18 @@ def detect_tokens(
         - BadFunctionCallOutput if a local node is used and the contract for the
           token has no code. That means the chain is not synced
         """
+        # Also detect tokens for proxies
+        addresses, proxies_mapping = self._proxies_for_addresses(addresses)
+
         with self.db.conn.read_ctx() as cursor:
             if only_cache is False:
                 self._detect_tokens(cursor, addresses=addresses)
 
             addresses_info: DetectedTokensType = {}
             for address in addresses:
-                addresses_info[address] = self.db.get_tokens_for_address(
+                # If `address` is a proxy, save token balances to its owner address
+                proxied_address = proxies_mapping.get(address, address)
```

Comment:
Isn't it a better way to do it? Here you are slowing a lot of the process if you have to check balances for one account instead of two :thinking: The aave module doesn't give you the token balances directly?

Reply:
Perhaps I am misunderstanding but why does it slow the process a lot?

Aave modules gives the balances but what if a proxy address not only has positions in aave but also owns some tokens? I think we should detect them

Reply:
proxies are not for storing tokens. They were designed to allow the execution of calls to multiple contracts in one transactions since EOA can't do that. If the token query is not needed ( I believe is not) I would completely skipt it. Take into consideration that we have over 3K+ tokens and that means that you have to make several calls to the detect which one is owned or not. If you have one account and a DSproxy you have to check for over 6K+ tokens and that increases the time spent doing the token detection.

Reply:
Hmmmm yeah, I know that ds proxy is designed to execute multiple calls but why at the end of the execution the proxy can't end up with having some tokens on its balance?

Reply:
the problem with this approach is that you are slowing down the entire token querying process by a geometric factor. The more tokens we add, the slower it will be, multiplied by the number of proxies you have ever used.

Protocols that use DSproxy tend to each make their own proxy. So I got multiple addresses doing DeFi stuff, each one of them having at least 4-5 proxies. If you detect all possible tokens and query balances for them we are dead.